File: 55_jump_game.cpp
class Solution {
public:
    bool canJump(vector<int>& nums) {
        int max_reach=nums[0];
        if(nums.size()==0) return false;
        
        
        
        for (int i=0; i<nums.size(); i++){
           
            if(i+nums[i]>max_reach){
                max_reach=i+nums[i];
            }
           
            if(max_reach==i){
                break;
            }
           
            
          
           }
        return max_reach>=nums.size()-1;
       
      
    }
};

File: 55_jump_game.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        max_reach=0
        if(len(nums)==1):
            return True
        
        for i in range(len(nums)):
           
            if(i+nums[i]>max_reach):
                max_reach=i+nums[i]
            if(max_reach==i):
                break
                
        if(max_reach>=len(nums)-1):
            return True
        else:
            return False
